File: backend/uploads/NoiseThreshold.m
function result = NoiseThreshold(channel_key, threshold)
    % Read JSON data from file based on channel_key
    % Assuming channel_key is the filename
    json_data = channel_key

    % Extract X_value and Y_value from the JSON data
    X_value = json_data.X_value;
    Y_value = json_data.Y_value;

    % Replace all Y_value entries with absolute value less than threshold to 0
    Y_value(abs(Y_value) < threshold) = 0;

    % Prepare JSON output
    result_struct.X_value = X_value;
    result_struct.Y_value = Y_value;
    result = jsonencode(result_struct);
end

File: backend/uploads/NoiseThreshold.py
import numpy as np
import json
def noise_threshold(channel_key, threshold):
    """
    Process JSON data by thresholding Y values
    
    Parameters:
    channel_key (dict or str): Either a dictionary containing the JSON data 
                              or a string with the JSON content
    threshold (float): Threshold value for Y values
    
    Returns:
    str: JSON string with processed data
    """
    # If channel_key is a string, parse it as JSON
    if isinstance(channel_key, str):
        json_data = json.loads(channel_key)
    else:
        json_data = channel_key
    
    # Extract X_value and Y_value
    X_value = json_data['X_value']
    Y_value = np.array(json_data['Y_value'])  # Convert to numpy array for vectorized operations
    
    # Replace all Y_value entries with absolute value less than threshold to 0
    Y_value[np.abs(Y_value) < threshold] = 0
    
    # Prepare output dictionary
    result_dict = {
        'X_value': X_value,
        'Y_value': Y_value.tolist()  # Convert back to list for JSON serialization
    }
    
    return json.dumps(result_dict)